### ISSUE TITLE
fertirigation mode

### ISSUE THREAD

**Issue by openspirulina** [OpenSprinkler/OpenSprinkler-Firmware#79]
add the possibility to one valve be a fertirrigation valve. 
the behavoir if this valve must be as a fertirragtion valve, so you should assing a feritirrigation time and a sector.

**Comment by rayshobby** [comment]
Can you explain what is the behavior of a fertirrigation valve?

**Comment by openspirulina** [comment]
Yes, I can.
fertirrigation constist to add some fertilizer to the water. Usually you have a tank of fertilizer and you add for a short period of time or for some liters of fertilzer during the normal cycle of the valve...
It can have each own flowmeter. 

This cycle of fertirrigation it's not recommended to be at the begging niether the end of the irrigation period of each sector, as the fertilizer could reamain inside the pipe...or not enough water inside the pipe and there not be a good mixture.

So the normal working is: you have to define the irragtion timetable (programe), and later you define the fertirrigation period for each program. 
The fertirrigation period will be smaller than the irrigation sector, and there should be the possibility to define "where" to start the ferritirrigation period. For exemple at the middle of the irrigation cylce, after 10minutes of starting the irrigation, or something like that.

The funcionallty is quite simple, the fertirrigation valve will be defined as valve number "X" (as you define the general valve) and will open the valve while fertirrigation period. (which means that at least general valve and 1 sector valve are open). 

I expect to be clear enough, please let me know if you have any other doubt.

**Comment by mollipolli** [comment]
Normally a fertigation valve is just another station.  OS in it's current form should have no issue with it, so long as the current draw from the valves is not too high.  It requires at least 3 valves using 2 stations.  At our nursery we utilize basically this setup for our greenhouses:

- At the house we have a barrel of concentrate mixed with an injector to a line.  Clear water comes in plumbed to a Tee.  One leg of the tee goes to a "Normally Closed" valve and the injector.  The other goes to a "Normally Open" Valve. Both legs are then combined after.

- In use you would actually wire both valves together onto one station.  When powered the Normally Open valve will close (clear) and the Normally Closed valve will open (fertilizer).

- A third valve is then used at the location to turn on and off the water, clear or fertilizer to the irrigation.

- If multiple locations are required to run on the same fertilizer source, you would then want different valves per location.  

This requires running at least 3 valves at once (2 at the source wired together on a station, and 1 at the location on a separate OS station).  You might have to watch the current not to overload OS, especially when the solenoids first power.  You could use delays or multiple OS units to get around that though.  Also check your water supply codes.  When you start putting fertilizer into irrigation there are extra requirements to make sure it doesn't back-flow into the source.

**Comment by openspirulina** [comment]
thank u mollipoli, 
I know that I can do fertirrigation whit OS right now. but, defining one valve as a fertilizer valve and it has the right behavoir, it would be much much easier to configure and to manage the fertiirgation process... when u have plenty of sector, and changing the irrigation and fertiirrigation periods with this method it becomes complicated and tired.. 

I do not have the knowledge to implemente this, but I feel that it'll be great if someone could do this, specialy for professionals farmers...

**Comment by daveperdue** [comment]
You can name the station Fertirrigation or whatever else you want to denote its purpose.

**Comment by openspirulina** [comment]
yes, i'm capable to do this daveperdue...but it's not an improvement.

p.e. i've 5 sector each one irrigation one hour, and fertirrigation 10 mintes. easy to configurate...(at least once)
if I increase up to 2 hours irrigation time per sector i've to reconfigurate for all the sectors fertirrigation valve...

instead of this if fertirragation is associated with one sector it will do the fertirrigation whenever this sector is open....(and no mistakes calculating times...)
actually, this is how normal comercial irrigation computers works...

**Comment by jbaudoux** [comment]
> * At the house we have a barrel of concentrate mixed with an injector to a line.  Clear water comes in plumbed to a Tee.  One leg of the tee goes to a "Normally Closed" valve and the injector.  The other goes to a "Normally Open" Valve. Both legs are then combined after.

@mollipolli I'm using venturi injector. I guess that for such setup you need to have the injector between 2 valves. Without a valve after, there will be a backward flow in the injector. And without a valve before, there is a forward flow in the injector. In both case, to much pressure is applied on the injector.
